src/control/lib/control/pool.go

```diff
@@ -608,25 +608,37 @@ func PoolQuery(ctx context.Context, rpcClient UnaryInvoker, req *PoolQueryReq) (
 	}
 
 	pqr := new(PoolQueryResp)
-	return pqr, convertMSResponse(ur, pqr)
+	err = convertMSResponse(ur, pqr)
+	if err != nil {
+		return nil, err
+	}
+
+	err = pqr.UpdateState()
+	if err != nil {
+		return nil, err
+	}
+
+	return pqr, err
 }
 
-func UpdatePoolQueryState(pqr *PoolQueryResp) {
+// Update the pool state
+func (pqr *PoolQueryResp) UpdateState() error {
 	// Pool state is unknown as default, if TotalTargets is 0.
 	if pqr.TotalTargets == 0 {
-		pqr.State = "Unknown"
+		pqr.State = system.PoolServiceStateUnknown.String()
```

Comment:
Not ideal having the state stored as a string, should really store the PoolServiceState itself, is this done because of difficulty with type conversion and JSON output compatibility?